Applications/Spire/Include/Spire/Spire/ToTextModel.hpp

```diff
@@ -0,0 +1,140 @@
+#ifndef SPIRE_TO_TEXT_MODEL_HPP
+#define SPIRE_TO_TEXT_MODEL_HPP
+#include <functional>
+#include <boost/optional/optional.hpp>
+#include "Spire/Spire/ValueModel.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+
+namespace Spire {
+
+  /**
+   * Implements a model that wraps a reference model and converts its values to
+   * and from their string representations.
+   * @param <T> The type of the reference model.
+   */
+  template<typename T>
+  class ToTextModel : public ValueModel<QString> {
+    public:
+
+      /** The type used by the reference model. */
+      using ModelType = T;
+
+      /** Function for converting a value to a QString. */
+      using ToString = std::function<QString (const ModelType& value)>;
+
+      /** Function for converting a QString to a value. */
+      using ToType =
+        std::function<boost::optional<ModelType> (const QString& value)>;
+
+      /**
+       * Constructs a ToTextModel with default conversion functions.
+       * @param model The reference model.
+       */
+      ToTextModel(std::shared_ptr<ValueModel<ModelType>> model);
+
+      /**
+       * Constructs a ToTextModel.
+       * @param model The reference model.
+       * @param to_string The value to QString conversion function.
+       */
+      ToTextModel(
+        std::shared_ptr<ValueModel<ModelType>> model, ToString to_string);
+
+      /**
+       * Constructs a ToTextModel.
+       * @param model The reference model.
+       * @param to_type The QString to value conversion function.
+       */
+      ToTextModel(
+        std::shared_ptr<ValueModel<ModelType>> model, ToType to_type);
+
+      /**
+       * Constructs a ToTextModel.
+       * @param model The reference model.
+       * @param to_string The value to QString conversion function.
+       * @param to_type The QString to value conversion function.
+       */
+      ToTextModel(
+        std::shared_ptr<ValueModel<ModelType>> model, ToString to_string,
+        ToType to_type);
+
+      QValidator::State get_state() const override;
+
+      const QString& get_current() const override;
+
+      QValidator::State set_current(const QString& value) override;
+
+      boost::signals2::connection connect_current_signal(
+        const typename CurrentSignal::slot_type& slot) const override;
+
+    private:
+      mutable CurrentSignal m_current_signal;
+      std::shared_ptr<ValueModel<ModelType>> m_model;
+      CustomVariantItemDelegate m_delegate;
```

Comment:
Remove this unused field.

Reply:
It's used in the default ToTextModel::to_string function. Should I just construct it as it's used?

Reply:
Nevermind, I see the comment below.